networkit/cpp/flow/EdmondsKarp.cpp

```diff
@@ -55,22 +56,125 @@ edgeweight EdmondsKarp::BFS(std::vector<edgeweight> &residFlow, std::vector<node
     return 0.0;
 }
 
+edgeweight EdmondsKarp::directedBFS(const std::vector<count> &reverseEdges,
+                                    std::vector<node> &pred) const {
+    std::fill(pred.begin(), pred.end(), none);
+    pred.resize(graph->upperNodeIdBound(), none);
+    std::vector<edgeweight> gain(graph->upperNodeIdBound(), std::numeric_limits<edgeweight>::max());
+
+    std::queue<node> Q;
+    Q.push(source);
+    pred[source] = source;
+    gain[source] = std::numeric_limits<edgeweight>::max();
+    ;
+
+    while (!Q.empty()) {
+        node u = Q.front();
+        Q.pop();
+        bool sinkReached = false;
+
+        graph->forNeighborsOf(u, [&](node, node v, edgeweight weight, edgeid eid) {
```

Comment:
This can be simplified with:
```cpp
for (node v : G->inNeighborRange(u))
  // ...
```
so that you do not have to check `sinkReached`.

Reply:
Actually that's not true, unfortunately neighbor iterators do not provide edge ids, so I'm afraid that we have to keep this loop as it is.